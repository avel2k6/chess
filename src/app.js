import watcher from './watcher';

const app = () => {
  const [initialChessColls, initialChessRows] = [[], {}];
  const rowLength = 10;
  const collLength = 10;

  for (let i = 0; i < collLength; i += 1) {
    initialChessColls.push('');
  }

  for (let i = 0; i < rowLength; i += 1) {
    initialChessRows[i] = initialChessColls;
  }

  const whiteChessmens = [
    { targets: ['7_1', '7_2', '7_3', '7_4', '7_5', '7_6', '7_7', '7_8'], figure: '♙' },
    { targets: ['8_1', '8_8'], figure: '♖' },
    { targets: ['8_2', '8_7'], figure: '♘' },
    { targets: ['8_3', '8_6'], figure: '♗' },
    { targets: ['8_4'], figure: '♕' },
    { targets: ['8_5'], figure: '♔' },
  ];

  const blackChessmens = [
    { targets: ['2_1', '2_2', '2_3', '2_4', '2_5', '2_6', '2_7', '2_8'], figure: '♟' },
    { targets: ['1_1', '1_8'], figure: '♜' },
    { targets: ['1_2', '1_7'], figure: '♞' },
    { targets: ['1_3', '1_6'], figure: '♝' },
    { targets: ['1_4'], figure: '♛' },
    { targets: ['1_5'], figure: '♚' },
  ];

  initialChessRows['9'] = [' ', 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', ' '];
  initialChessRows['8'] = ['1', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', '1'];
  initialChessRows['7'] = ['2', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', '2'];
  initialChessRows['6'] = ['3', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', '3'];
  initialChessRows['5'] = ['4', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', '4'];
  initialChessRows['4'] = ['5', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', '5'];
  initialChessRows['3'] = ['6', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', '6'];
  initialChessRows['2'] = ['7', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', '7'];
  initialChessRows['1'] = ['8', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', '8'];
  initialChessRows['0'] = [' ', 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', ' '];

  const appState = {
    chessFigures: {},
    chessField: [],
    checkedItem: {},
    targetItem: {},
    action: 'getItem',
    movedToPosition: {},
  };

  const controls = (formCurrentState) => {
    const state = formCurrentState;


    const formInput = document.querySelector('#chess');
    formInput.addEventListener(
      'click',
      (e) => {
        const { target } = e;
        const tagetClasses = target.classList;
        if (!tagetClasses.contains('borders') && tagetClasses.contains('item')) {
          const elText = e.target.textContent;
          const elId = e.target.id;
          if (state.action === 'getItem') {
            if (elText === ' ') { // Не даем брать 'пустые' поля
              return;
            }
            state.checkedItem = {
              elText,
              elId,
            };
            state.action = 'setItem';
            return;
          }

          if (state.action === 'setItem') {
            state.targetItem = {
              elText,
              elId,
            };
            state.action = 'getItem';
          }
        }
      },
    );

    const figuresInitialPosition = document.querySelectorAll('.figures');
    figuresInitialPosition.forEach(
      (initialPosition) => {
        initialPosition.addEventListener(
          'click',
          (e) => {
            const { target } = e;
            if (target.classList.contains('item-initial')) {
              const { textContent } = target;
              const targetPostition = target.getAttribute('target');
              state.movedToPosition = { targetPostition, textContent };
            }
          },
        );
      },
    );
  };

  controls(appState);
  watcher(appState);

  appState.chessField = initialChessRows;
  appState.chessFigures = {
    whiteChessmens,
    blackChessmens,
  };
};

export default app;
