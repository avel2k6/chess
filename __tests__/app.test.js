import '@babel/polyfill';
import fs from 'fs';
import path from 'path';
import timer from 'timer-promise';
import { html } from 'js-beautify';

import app from '../src/app';

const fixuturesPath = path.join(__dirname, '__fixtures__');
const getTree = () => html(document.body.innerHTML);


beforeEach(() => {
  const initHtml = fs.readFileSync(path.join(fixuturesPath, 'index.html')).toString();
  document.documentElement.innerHTML = initHtml;
  app();
});

test('application', async () => {
  expect(getTree()).toMatchSnapshot();

  await timer.start(200);
  expect(getTree()).toMatchSnapshot();
});
