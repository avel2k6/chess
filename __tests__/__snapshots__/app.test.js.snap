// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`application 1`] = `
"<div class=\\"container\\">
    <div id=\\"blackChessmens\\" class=\\"figures\\">
        Loading black...
    </div>
    <div class=\\"separator\\">
    </div>
    <div id=\\"chess\\" class=\\"chessfield\\">
        <div id=\\"loading\\">
            Loading chess...
        </div>
    </div>
    <div class=\\"separator\\">
    </div>
    <div id=\\"whiteChessmens\\" class=\\"figures\\">
        Loading white...
    </div>
</div>
<script type=\\"text/javascript\\" src=\\"/main.js\\"></script>"
`;

exports[`application 2`] = `
"<div class=\\"container\\">
    <div id=\\"blackChessmens\\" class=\\"figures\\">
        <table>
            <tr>
                <td target=\\"2_1\\" class=\\"item-initial\\">♟</td>
                <td target=\\"2_2\\" class=\\"item-initial\\">♟</td>
                <td target=\\"2_3\\" class=\\"item-initial\\">♟</td>
                <td target=\\"2_4\\" class=\\"item-initial\\">♟</td>
                <td target=\\"2_5\\" class=\\"item-initial\\">♟</td>
                <td target=\\"2_6\\" class=\\"item-initial\\">♟</td>
                <td target=\\"2_7\\" class=\\"item-initial\\">♟</td>
                <td target=\\"2_8\\" class=\\"item-initial\\">♟</td>
                <td target=\\"1_1\\" class=\\"item-initial\\">♜</td>
                <td target=\\"1_8\\" class=\\"item-initial\\">♜</td>
                <td target=\\"1_2\\" class=\\"item-initial\\">♞</td>
                <td target=\\"1_7\\" class=\\"item-initial\\">♞</td>
                <td target=\\"1_3\\" class=\\"item-initial\\">♝</td>
                <td target=\\"1_6\\" class=\\"item-initial\\">♝</td>
                <td target=\\"1_4\\" class=\\"item-initial\\">♛</td>
                <td target=\\"1_5\\" class=\\"item-initial\\">♚</td>
            </tr>
        </table>
    </div>
    <div class=\\"separator\\">
    </div>
    <div id=\\"chess\\" class=\\"chessfield\\">

        <table>
            <tr>
                <td class=\\"borders\\" id=\\"0_0\\"> </td>
                <td class=\\"borders\\" id=\\"0_1\\">a</td>
                <td class=\\"borders\\" id=\\"0_2\\">b</td>
                <td class=\\"borders\\" id=\\"0_3\\">c</td>
                <td class=\\"borders\\" id=\\"0_4\\">d</td>
                <td class=\\"borders\\" id=\\"0_5\\">e</td>
                <td class=\\"borders\\" id=\\"0_6\\">f</td>
                <td class=\\"borders\\" id=\\"0_7\\">g</td>
                <td class=\\"borders\\" id=\\"0_8\\">h</td>
                <td class=\\"borders\\" id=\\"0_9\\"> </td>
            </tr>
            <tr>
                <td class=\\"borders\\" id=\\"1_0\\">8</td>
                <td class=\\"item\\" id=\\"1_1\\"> </td>
                <td class=\\"item\\" id=\\"1_2\\"> </td>
                <td class=\\"item\\" id=\\"1_3\\"> </td>
                <td class=\\"item\\" id=\\"1_4\\"> </td>
                <td class=\\"item\\" id=\\"1_5\\"> </td>
                <td class=\\"item\\" id=\\"1_6\\"> </td>
                <td class=\\"item\\" id=\\"1_7\\"> </td>
                <td class=\\"item\\" id=\\"1_8\\"> </td>
                <td class=\\"borders\\" id=\\"1_9\\">8</td>
            </tr>
            <tr>
                <td class=\\"borders\\" id=\\"2_0\\">7</td>
                <td class=\\"item\\" id=\\"2_1\\"> </td>
                <td class=\\"item\\" id=\\"2_2\\"> </td>
                <td class=\\"item\\" id=\\"2_3\\"> </td>
                <td class=\\"item\\" id=\\"2_4\\"> </td>
                <td class=\\"item\\" id=\\"2_5\\"> </td>
                <td class=\\"item\\" id=\\"2_6\\"> </td>
                <td class=\\"item\\" id=\\"2_7\\"> </td>
                <td class=\\"item\\" id=\\"2_8\\"> </td>
                <td class=\\"borders\\" id=\\"2_9\\">7</td>
            </tr>
            <tr>
                <td class=\\"borders\\" id=\\"3_0\\">6</td>
                <td class=\\"item\\" id=\\"3_1\\"> </td>
                <td class=\\"item\\" id=\\"3_2\\"> </td>
                <td class=\\"item\\" id=\\"3_3\\"> </td>
                <td class=\\"item\\" id=\\"3_4\\"> </td>
                <td class=\\"item\\" id=\\"3_5\\"> </td>
                <td class=\\"item\\" id=\\"3_6\\"> </td>
                <td class=\\"item\\" id=\\"3_7\\"> </td>
                <td class=\\"item\\" id=\\"3_8\\"> </td>
                <td class=\\"borders\\" id=\\"3_9\\">6</td>
            </tr>
            <tr>
                <td class=\\"borders\\" id=\\"4_0\\">5</td>
                <td class=\\"item\\" id=\\"4_1\\"> </td>
                <td class=\\"item\\" id=\\"4_2\\"> </td>
                <td class=\\"item\\" id=\\"4_3\\"> </td>
                <td class=\\"item\\" id=\\"4_4\\"> </td>
                <td class=\\"item\\" id=\\"4_5\\"> </td>
                <td class=\\"item\\" id=\\"4_6\\"> </td>
                <td class=\\"item\\" id=\\"4_7\\"> </td>
                <td class=\\"item\\" id=\\"4_8\\"> </td>
                <td class=\\"borders\\" id=\\"4_9\\">5</td>
            </tr>
            <tr>
                <td class=\\"borders\\" id=\\"5_0\\">4</td>
                <td class=\\"item\\" id=\\"5_1\\"> </td>
                <td class=\\"item\\" id=\\"5_2\\"> </td>
                <td class=\\"item\\" id=\\"5_3\\"> </td>
                <td class=\\"item\\" id=\\"5_4\\"> </td>
                <td class=\\"item\\" id=\\"5_5\\"> </td>
                <td class=\\"item\\" id=\\"5_6\\"> </td>
                <td class=\\"item\\" id=\\"5_7\\"> </td>
                <td class=\\"item\\" id=\\"5_8\\"> </td>
                <td class=\\"borders\\" id=\\"5_9\\">4</td>
            </tr>
            <tr>
                <td class=\\"borders\\" id=\\"6_0\\">3</td>
                <td class=\\"item\\" id=\\"6_1\\"> </td>
                <td class=\\"item\\" id=\\"6_2\\"> </td>
                <td class=\\"item\\" id=\\"6_3\\"> </td>
                <td class=\\"item\\" id=\\"6_4\\"> </td>
                <td class=\\"item\\" id=\\"6_5\\"> </td>
                <td class=\\"item\\" id=\\"6_6\\"> </td>
                <td class=\\"item\\" id=\\"6_7\\"> </td>
                <td class=\\"item\\" id=\\"6_8\\"> </td>
                <td class=\\"borders\\" id=\\"6_9\\">3</td>
            </tr>
            <tr>
                <td class=\\"borders\\" id=\\"7_0\\">2</td>
                <td class=\\"item\\" id=\\"7_1\\"> </td>
                <td class=\\"item\\" id=\\"7_2\\"> </td>
                <td class=\\"item\\" id=\\"7_3\\"> </td>
                <td class=\\"item\\" id=\\"7_4\\"> </td>
                <td class=\\"item\\" id=\\"7_5\\"> </td>
                <td class=\\"item\\" id=\\"7_6\\"> </td>
                <td class=\\"item\\" id=\\"7_7\\"> </td>
                <td class=\\"item\\" id=\\"7_8\\"> </td>
                <td class=\\"borders\\" id=\\"7_9\\">2</td>
            </tr>
            <tr>
                <td class=\\"borders\\" id=\\"8_0\\">1</td>
                <td class=\\"item\\" id=\\"8_1\\"> </td>
                <td class=\\"item\\" id=\\"8_2\\"> </td>
                <td class=\\"item\\" id=\\"8_3\\"> </td>
                <td class=\\"item\\" id=\\"8_4\\"> </td>
                <td class=\\"item\\" id=\\"8_5\\"> </td>
                <td class=\\"item\\" id=\\"8_6\\"> </td>
                <td class=\\"item\\" id=\\"8_7\\"> </td>
                <td class=\\"item\\" id=\\"8_8\\"> </td>
                <td class=\\"borders\\" id=\\"8_9\\">1</td>
            </tr>
            <tr>
                <td class=\\"borders\\" id=\\"9_0\\"> </td>
                <td class=\\"borders\\" id=\\"9_1\\">a</td>
                <td class=\\"borders\\" id=\\"9_2\\">b</td>
                <td class=\\"borders\\" id=\\"9_3\\">c</td>
                <td class=\\"borders\\" id=\\"9_4\\">d</td>
                <td class=\\"borders\\" id=\\"9_5\\">e</td>
                <td class=\\"borders\\" id=\\"9_6\\">f</td>
                <td class=\\"borders\\" id=\\"9_7\\">g</td>
                <td class=\\"borders\\" id=\\"9_8\\">h</td>
                <td class=\\"borders\\" id=\\"9_9\\"> </td>
            </tr>
        </table>
    </div>
    <div class=\\"separator\\">
    </div>
    <div id=\\"whiteChessmens\\" class=\\"figures\\">
        <table>
            <tr>
                <td target=\\"7_1\\" class=\\"item-initial\\">♙</td>
                <td target=\\"7_2\\" class=\\"item-initial\\">♙</td>
                <td target=\\"7_3\\" class=\\"item-initial\\">♙</td>
                <td target=\\"7_4\\" class=\\"item-initial\\">♙</td>
                <td target=\\"7_5\\" class=\\"item-initial\\">♙</td>
                <td target=\\"7_6\\" class=\\"item-initial\\">♙</td>
                <td target=\\"7_7\\" class=\\"item-initial\\">♙</td>
                <td target=\\"7_8\\" class=\\"item-initial\\">♙</td>
                <td target=\\"8_1\\" class=\\"item-initial\\">♖</td>
                <td target=\\"8_8\\" class=\\"item-initial\\">♖</td>
                <td target=\\"8_2\\" class=\\"item-initial\\">♘</td>
                <td target=\\"8_7\\" class=\\"item-initial\\">♘</td>
                <td target=\\"8_3\\" class=\\"item-initial\\">♗</td>
                <td target=\\"8_6\\" class=\\"item-initial\\">♗</td>
                <td target=\\"8_4\\" class=\\"item-initial\\">♕</td>
                <td target=\\"8_5\\" class=\\"item-initial\\">♔</td>
            </tr>
        </table>
    </div>
</div>
<script type=\\"text/javascript\\" src=\\"/main.js\\"></script>"
`;
